packages/website/src/bin/page-generator/index.js

```diff
@@ -380,10 +380,19 @@ module.exports = async function generatePages(
 ) {
   cleanPages(pagesPath, docsList);
 
-  const packageInfo = getPackageInfo(packagesPaths, {
-    useManifests: options.useManifests,
-    showSubExamples: options.showSubExamples,
-  });
+  const pkgOpts = {};
+
+  if (typeof options.useManifests !== 'undefined') {
+    pkgOpts.useManifests = options.useManifests;
+  }
+  if (typeof options.showSubExamples !== 'undefined') {
+    pkgOpts.showSubExamples = options.showSubExamples;
+  }
+  if (typeof options.showExamples !== 'undefined') {
+    pkgOpts.showExamples = options.showExamples;
+  }
+
+  const packageInfo = getPackageInfo(packagesPaths, pkgOpts);
```

Comment:
Do we need to update generate-pages.js to pass in `showExamples` as well? Otherwise `showExamples` will always be undefined.

Might be worthwhile adding a test file for generate-pages.js or maybe some more integration test examples (probably in a subsequent PR).

Reply:
Also, not something for this PR but I think we also desperately need to type our config somehow and perform some runtime config validation, it's quite unclear at the moment what config options are available.

[Stricter](https://github.com/stricter/stricter/blob/master/src/config/validate-config.ts) achieves this by generating a json schema from its typescript types and validating the user config using `ajv`, we could possibly do something similar, although I noticed we aren't writing our bin files as ts which I think may be a good idea in the future?

Reply:
Yeah, we want to:
convert everything to TS
strongly type our config

We should probably make tickets for it.